storage/inmem/inmem.go

```diff
@@ -92,6 +92,79 @@ func (db *store) NewTransaction(_ context.Context, params ...storage.Transaction
 	return newTransaction(xid, write, context, db), nil
 }
 
+// Truncate implements the storage.Store interface. This method must be called within a transaction.
+func (db *store) Truncate(ctx context.Context, txn storage.Transaction, _ storage.TransactionParams, it storage.Iterator) error {
+	var update *storage.Update
+	var err error
+
+	underlying, err := db.underlying(txn)
+	if err != nil {
+		return err
+	}
+
+	for {
+		update, err = it.Next()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return err
+		}
```

Comment:
💭 Isn't this a bit redundant? If we did
```go
if err != nil {
  break
}
```
here, the check below would take care of handling io.EOF: https://github.com/open-policy-agent/opa/pull/4661/files#diff-871d6cd4f1861ef162eed55afc77054ca2274b7f967f9e5fdf298a97e7e7f154R161-R165 -- but I might be missing something 🤔

Reply:
Good catch. Updated.